v2/api/network/customizations/route_table_extensions.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package customizations
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101storage"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+// Attention: A lot of code in this file is very similar to the logic in virtual_network_extensions.go.
+// The two should be kept in sync as much as possible.
+
+var _ extensions.ARMResourceModifier = &RouteTableExtension{}
+
+func (extension *RouteTableExtension) ModifyARMResource(
+	ctx context.Context,
+	armObj genruntime.ARMResource,
+	obj genruntime.ARMMetaObject,
+	kubeClient kubeclient.Client,
+	resolver *resolver.Resolver,
+	log logr.Logger) (genruntime.ARMResource, error) {
+
+	typedObj, ok := obj.(*network.RouteTable)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T", obj)
+	}
+
+	// Type assert that we are the hub type. This should fail to compile if
```

Comment:
```suggestion
	// Type assert that we are the hub type. This serves to fail the build if
```

Reply:
I changed this to `will fail to compile` rather than `serves to fail the build`. I _think_ that's moving in the direction you were getting at (using "should" was weird there) but I like that wording more.